src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -1752,9 +1752,21 @@ bool RBMCleaner::equals(const RBMSpaceTracker &_other) const
                          state == rbm_extent_state_t::RESERVED))) {
       // pass
     } else {
-      all_match = false;
-      ERROR("block addr {} mismatch other used: {}",
-	    addr, i->second.used);
+      paddr_t paddr = convert_abs_addr_to_paddr(addr, rbm->get_device_id());
+      auto entries = backref_manager.get_cached_backref_extents_in_range(
+	paddr, paddr.add_offset(rbm->get_block_size()));
+      bool match = false;
+      for (auto p : entries) {
+	if (p.paddr == paddr) {
+	  match = true;
+	  break;
+	}
+      }
+      if (!match) {
+	all_match = false;
+	ERROR("block addr {} mismatch other used: {}",
+	      addr, i->second.used);
+      }
```

Comment:
Why is this necessary?

IIUC this logic implies that the rbm allocator may not contain the allocations of the backref-tree nodes.

But it doesn't match my understanding that `epm.commit/mark_space_used` are including backref-tree nodes in both transaction commit and replay paths, so the rbm allocator should contain the allocations from the backref-tree nodes.

Or am I still missing anything?

Reply:
`other`is the result (RBMSpaceTracker) from `scan_mapped_space`---therefore, it contains current backref tree's information about block allocation. At this point, ExtentAllocator(RBM) keeps track of the allocations, but the backref tree's result does not because alloc-deltas are not yet replayed. So, this part checks if the address is allocated for backref node. Am I missing?

Reply:
Sorry, I was mistaken about error log I captured before. Removed.